src/services/agents/coach_agent.py

```
@@ -0,0 +1,219 @@
+
+    JSON Output:
+    """
+    try:
+        response = openai_client.chat.completions.create(
+            model="gpt-4o",
+            messages=[{"role": "user", "content": prompt}],
+            response_format={"type": "json_object"},
+        )
+
+        if not response.choices or not response.choices[0].message.content:
+            raise ValueError("Received an invalid or empty response from OpenAI API.")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)